src/stories/temp-converter.tsx

```diff
@@ -0,0 +1,105 @@
+            return ((value - 32) * 5) / 9;
+        if (from === "Fahrenheit" && to === "Kelvin")
+            return ((value - 32) * 5) / 9 + 273.15;
+        if (from === "Kelvin" && to === "Celsius") return value - 273.15;
+        if (from === "Kelvin" && to === "Fahrenheit")
+            return ((value - 273.15) * 9) / 5 + 32;
+
+        return NaN;
+    };
+
+    useEffect(() => {
+        if (!isDisabled) {
+            const result = convertTemperature(
+                parseFloat(temperature),
+                startUnit,
+                endUnit
+            );
+            setConvertedTemp(result.toFixed(2));
+        } else {
+            setConvertedTemp("");
+        }
+    }, [temperature, startUnit, endUnit, isDisabled]);
+
+    return (
+        <div>
+            <p className="text-2xl">Temperature Converter</p>
+            <p>Enter the temperature, select units and submit.</p>
+            <div className="flex gap-4">
+                <input
+                    type="text"
+                    placeholder="0.00"
+                    value={temperature}
+                    onChange={(e) => setTemperature(e.target.value)}
+                />
+                <select
+                    value={startUnit}
+                    onChange={(e) => setStartUnit(e.target.value)}
+                >
+                    <option value="">Select Unit</option>
+                    {props.items
+                        .filter((unit) => unit !== endUnit)
+                        .map((unit) => (
+                            <option key={unit} value={unit}>
+                                {unit}
+                            </option>
+                        ))}
+                </select>
+                <select
+                    value={endUnit}
+                    onChange={(e) => setEndUnit(e.target.value)}
+                >
+                    <option value="">Select Unit</option>
+                    {props.items
+                        .filter((unit) => unit !== startUnit)
+                        .map((unit) => (
+                            <option key={unit} value={unit}>
+                                {unit}
+                            </option>
+                        ))}
+                </select>
+                <button
+                    disabled={isDisabled}
+                    className={`text-white text-lg rounded-xl p-2 px-6 cursor-pointer ${
+                        isDisabled ? "bg-gray-500" : "bg-black"
+                    }`}
+                    onClick={() => {
+                        const result = convertTemperature(
+                            parseFloat(temperature),
+                            startUnit,
+                            endUnit
+                        );
+                        setConvertedTemp(result.toFixed(2));
+                    }}
+                >
+                    Convert
+                </button>
+            </div>
+            <p>
```

Comment:
It would be better to not render the `<p>` element in general instead of just making its content conditionally rendered.  
Generally: Only render what's necessary.